src/eos-updater.c

```diff
@@ -117,6 +149,77 @@ on_name_lost (GDBusConnection *connection,
   message ("Lost the name %s\n", name);
 }
 
+static gchar *
+quit_file_name (void)
+{
+  return eos_updater_dup_envvar_or ("EOS_UPDATER_TEST_UPDATER_QUIT_FILE",
+                                     NULL);
+}
+
+static EosQuitFileCheckResult
+check_and_quit (gpointer local_data_ptr)
+{
+  LocalData *local_data = local_data_ptr;
+  EosUpdaterState state;
+
+  state = eos_updater_get_state (local_data->updater);
+  switch (state)
+    {
+    case EOS_UPDATER_STATE_NONE:
+    case EOS_UPDATER_STATE_READY:
+    case EOS_UPDATER_STATE_ERROR:
+    case EOS_UPDATER_STATE_UPDATE_APPLIED:
+      g_main_loop_quit (local_data->loop);
+      return EOS_QUIT_FILE_QUIT;
+
+    case EOS_UPDATER_STATE_POLLING:
+    case EOS_UPDATER_STATE_UPDATE_AVAILABLE:
+    case EOS_UPDATER_STATE_FETCHING:
+    case EOS_UPDATER_STATE_UPDATE_READY:
+    case EOS_UPDATER_STATE_APPLYING_UPDATE:
+      return EOS_QUIT_FILE_KEEP_CHECKING;
+
+    case EOS_UPDATER_N_STATES:
```

Comment:
Nah, I prefer gcc to yell at me about unhandled enum member, especially for enums that we have a full control over.